src/main/java/address/ui/Ui.java

```diff
@@ -0,0 +1,34 @@
+package address.ui;
+
+import address.MainApp;
+import address.browser.BrowserManager;
+import address.controller.MainController;
+import address.model.ModelManager;
+import address.util.Config;
+import javafx.scene.control.Alert;
+import javafx.stage.Stage;
+
+/**
+ * The UI of the app.
+ */
+public class Ui {
+    MainController mainController;
+
+    public Ui(MainApp mainApp, ModelManager modelManager, Config config){
+        BrowserManager.initializeJxBrowserEnvironment();
```

Comment:
Java WebView can't be initialized in non-ui thread. Another way will be to have two init method in the browserManager. one for calling in UI thread and one not. I guess this will complicate stuffs.
________________________________
From: Hans Adrian <notifications@github.com>
Sent: Monday, June 27, 2016 11:41:28 AM
To: HubTurbo/addressbook
Cc: Lim You Liang; Mention
Subject: Re: [HubTurbo/addressbook] Added a UI component to front all ui-related code (#299)


In src/main/java/address/ui/Ui.java<https://github.com/HubTurbo/addressbook/pull/299#discussion_r68519508>:

> +import address.MainApp;
> +import address.browser.BrowserManager;
> +import address.controller.MainController;
> +import address.model.ModelManager;
> +import address.util.Config;
> +import javafx.scene.control.Alert;
> +import javafx.stage.Stage;
> +
> +/**
> + * The UI of the app.
> + */
> +public class Ui {
> +    MainController mainController;
> +
> +    public Ui(MainApp mainApp, ModelManager modelManager, Config config){
> +        BrowserManager.initializeJxBrowserEnvironment();


I only tried the initialization of jxBrowser, which only works in MainApp init method so what has been suggested actually is possible. Not sure about WebView as I was not involved.

-
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/HubTurbo/addressbook/pull/299/files/4e7dbc3324816c66d4e5655c086244cbe2d1168e#r68519508>, or mute the thread<https://github.com/notifications/unsubscribe/AIpoEBVGr8ucnBo4nv8tLzL960R38wIKks5qP0ZogaJpZM4I-Qes>.